core/server/proxy/src/main/java/alluxio/proxy/AlluxioProxyProcess.java

```diff
@@ -51,6 +61,9 @@ public final class AlluxioProxyProcess implements ProxyProcess {
 
   private final CountDownLatch mLatch;
 
+  private ExecutorService mPool = Executors.newFixedThreadPool(1,
```

Comment:
why not use ScheduledExecutorService here? I feel that implementation will be easier

Reply:
I chose this way to stay consistent with the register-heartbeat in worker
https://github.com/Alluxio/alluxio/blob/4ad6185ab23a7cf151e4289eb7e11b34f87f2d91/core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockWorker.java#L255